js/transform.js

```diff
@@ -222,6 +232,15 @@ function displayForm(e) {
 }
 
 function generateAndDisplayForm(formId, dest) {
+    const INTEGER_AFTER_UNDERSCORE_REGEX = /(.*)_([0-9]+)/;
+    const match = formId.match(INTEGER_AFTER_UNDERSCORE_REGEX);
+    if (match) {
```

Comment:
[Pebble] If I'm understanding the code correctly, even if there's no match, the method `.match` returns an array with an empty string `[""]`, which is actually "truthy" and would satisfy this `if` statement. A safer check here would be whether the first element of the array is not an empty string. I think being as explicit as possible is best, so `if (match.length > 0 && match[0].length > 0)`. You may also want to rename the var to `matches` to make that clearer. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Reply:
Great catch, thank you!

Reply:
@jachan Wait I completely misread the documentation, `.match` does actually return `null` if there's no match. SORRY! I take this back, you could keep it as you had it, but ideally make it more explicit to `if (match != null)`

Reply:
Thanks for catching that the catch was incorrect! ;)

Reply:
Glad y'all had this convo because I didnt realize .match returns null when there's no matches :)